### PR TITLE
Clean up auth code in preparation for adding SAML

### PR DESCRIPTION
**Pull request by gnprice**
These are all pure refactors, or tests.

They came as I was refactoring this code to make it suitable to incorporate the new data-driven list of auth methods (#3670) which we'll use for SAML... and then as I studied it harder and unraveled what some parts of it were really doing, and sought to make the names and organization better match that.

### CODE REVIEW

File: src/start/AuthScreen.js [hunk 1]
```diff
@@ -19,52 +19,52 @@ import { generateOtp, openBrowser, closeBrowser } from './oauth';
 import { loginSuccess, navigateToDev, navigateToPassword } from '../actions';
 
 type AuthenticationMethodDetails = {|
-  method: string,
   name: string,
+  displayName: string,
   Icon: IconType,
   handler: string,
 |};
```
Comment (ray-kraesig) [inline src/start/AuthScreen.js]: Given the commit message, it sounds like `'id'` or `'methodName'` might be a better key than `'name'`.

(Either way, these two items could use docstrings, or at least docstringishly descriptive comments.)

File: src/utils/__tests__/encoding-test.js [hunk 2]
```diff
@@ -78,11 +77,3 @@ describe('base64Utf8Encode', () => {
...
   });
 });
-describe('extractApiKey', () => {
-  test('correctly extracts an API key that has been XORed with a OTP', () => {
-    const key = 'testing';
-    const otp = 'A8348A93A83493';
-    const encoded = xorHexStrings(asciiToHex(key), otp);
-    expect(extractApiKey(encoded, otp)).toBe(key);
-  });
-});
```
Comment (ray-kraesig) [inline src/utils/__tests__/encoding-test.js]: This test was removed, but doesn't seem to have been replaced in a new location.

File: src/start/AuthScreen.js (hunk 1 above)
Comment (gnprice) [reply src/start/AuthScreen.js]: Yeah, I agree `name` is ambiguous on identifier-like name vs. display name. I think it helps that the other name in the pair is unambiguous, though.

To me `id` would suggest a numeric ID -- like our user IDs, realm IDs, stream IDs, etc. Realms and streams have identifier-like names, even, but then also "IDs" which are numbers. I think `methodName` doesn't add any information beyond `name`; to the extent an auth method's "name" could mean its display name, "method name" could just as well. So I don't think either of those would help.

Some jsdoc is a good idea. I'll add some.

File: src/utils/__tests__/encoding-test.js (hunk 2 above)
Comment (gnprice) [reply src/utils/__tests__/encoding-test.js]: Thanks. That was intentional, but I should mention it in the commit message. (The test just recites the implementation, and I didn't see value in keeping it. Once `authFromCallbackUrl` gets some tests a couple of commits later, that effectively includes a new test of this code that actually specifies the input.)

File: src/start/AuthScreen.js (hunk 1 above)
Comment (ray-kraesig) [reply src/start/AuthScreen.js]: > To me `id` would suggest a numeric ID -- like our user IDs, realm IDs, stream IDs, etc.

Ah, that's understandable. And noted. (The meaning I was thinking of was from the DOM:  `document.getElementById('blorple')`, etc.)

**Comment by gnprice** [comment]
Thanks @ray-kraesig for the review! Merged, with revisions from your comments.

File: src/start/AuthScreen.js (hunk 1 above)
Comment (gnprice) [reply src/start/AuthScreen.js]: > (The meaning I was thinking of was from the DOM: `document.getElementById('blorple')`, etc.)

Ah, makes sense! Yeah, within Zulip I think we pretty consistently say "ID" only for a numeric ID.

The one exception I can think of is within the server codebase -- we've managed to keep it out of the API -- where `Realm.string_id` is an identifier-style string, better known as the "subdomain".

**Pull request merged**